<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>s21::MainWindow</class>
 <widget class="QMainWindow" name="s21::MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>996</width>
    <height>927</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="s21::centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="3" column="2">
     <widget class="QDoubleSpinBox" name="green_edge_sbox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
    </item>
    <item row="5" column="4">
     <widget class="QDoubleSpinBox" name="z_rot_sbox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimum">
       <double>-0.500000000000000</double>
      </property>
      <property name="maximum">
       <double>0.500000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QPushButton" name="file_browse_btn">
      <property name="text">
       <string>Browse .obj file:</string>
      </property>
     </widget>
    </item>
    <item row="9" column="0" colspan="6">
     <widget class="Render" name="openGLWidget">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item row="5" column="2">
     <widget class="QDoubleSpinBox" name="y_rot_sbox">
      <property name="minimum">
       <double>-0.500000000000000</double>
      </property>
      <property name="maximum">
       <double>0.500000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="width_edge_btn">
      <property name="text">
       <string>Width of edges</string>
      </property>
     </widget>
    </item>
    <item row="7" column="4" colspan="2">
     <widget class="QPushButton" name="central_proj_btn">
      <property name="text">
       <string>Central Projection</string>
      </property>
     </widget>
    </item>
    <item row="4" column="3">
     <widget class="QPushButton" name="y_move_btn">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>yMove</string>
      </property>
     </widget>
    </item>
    <item row="4" column="5">
     <widget class="QPushButton" name="z_move_btn">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>zMove</string>
      </property>
     </widget>
    </item>
    <item row="5" column="3">
     <widget class="QPushButton" name="y_rot_btn">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>yRotation</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1" colspan="3">
     <widget class="QLabel" name="file_name">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QDoubleSpinBox" name="red_edge_sbox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QPushButton" name="red_bg">
      <property name="text">
       <string>Red BG</string>
      </property>
     </widget>
    </item>
    <item row="8" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>Count of vertexes:</string>
      </property>
     </widget>
    </item>
    <item row="3" column="4">
     <widget class="QDoubleSpinBox" name="blue_edge_sbox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
    </item>
    <item row="1" column="2" colspan="2">
     <widget class="QPushButton" name="dashed_btn">
      <property name="text">
       <string>Dashed lines</string>
      </property>
     </widget>
    </item>
    <item row="4" column="2">
     <widget class="QDoubleSpinBox" name="y_move_sbox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimum">
       <double>-0.500000000000000</double>
      </property>
      <property name="maximum">
       <double>0.500000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
    </item>
    <item row="7" column="0" colspan="2">
     <widget class="QPushButton" name="parallel_proj_btn">
      <property name="text">
       <string>Parallel Projection</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QDoubleSpinBox" name="red_clr_sbox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QDoubleSpinBox" name="width_edge_sbox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="singleStep">
       <double>0.500000000000000</double>
      </property>
     </widget>
    </item>
    <item row="3" column="5">
     <widget class="QPushButton" name="blue_edge_btn">
      <property name="text">
       <string>RGB3 Edges</string>
      </property>
     </widget>
    </item>
    <item row="8" column="4">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>Count of edges</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QDoubleSpinBox" name="x_rot_sbox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimum">
       <double>-0.500000000000000</double>
      </property>
      <property name="maximum">
       <double>0.500000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
    </item>
    <item row="4" column="4">
     <widget class="QDoubleSpinBox" name="z_move_sbox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimum">
       <double>-0.500000000000000</double>
      </property>
      <property name="maximum">
       <double>0.500000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
    </item>
    <item row="2" column="5">
     <widget class="QPushButton" name="blue_bg">
      <property name="text">
       <string>Blue BG</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QPushButton" name="x_move_btn">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>xMove</string>
      </property>
     </widget>
    </item>
    <item row="5" column="1">
     <widget class="QPushButton" name="x_rot_btn">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>xRotation</string>
      </property>
     </widget>
    </item>
    <item row="2" column="4">
     <widget class="QDoubleSpinBox" name="blue_clr_sbox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QDoubleSpinBox" name="green_clr_sbox">
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
    </item>
    <item row="3" column="3">
     <widget class="QPushButton" name="green_edge_btn">
      <property name="text">
       <string>RGB2 Edges</string>
      </property>
     </widget>
    </item>
    <item row="5" column="5">
     <widget class="QPushButton" name="z_rot_btn">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>zRotation</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QDoubleSpinBox" name="x_move_sbox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimum">
       <double>-0.500000000000000</double>
      </property>
      <property name="maximum">
       <double>0.500000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QPushButton" name="red_edge_btn">
      <property name="text">
       <string>RGB1 Edges</string>
      </property>
     </widget>
    </item>
    <item row="2" column="3">
     <widget class="QPushButton" name="green_bg">
      <property name="text">
       <string>Green BG</string>
      </property>
     </widget>
    </item>
    <item row="1" column="4" colspan="2">
     <widget class="QPushButton" name="straight_btn">
      <property name="text">
       <string>Straight lines</string>
      </property>
     </widget>
    </item>
    <item row="6" column="3" colspan="2">
     <widget class="QPushButton" name="zoom_out_btn">
      <property name="text">
       <string>Zoom Out</string>
      </property>
     </widget>
    </item>
    <item row="6" column="1" colspan="2">
     <widget class="QPushButton" name="zoom_in_btn">
      <property name="text">
       <string>Zoom In</string>
      </property>
     </widget>
    </item>
    <item row="8" column="1">
     <widget class="QLabel" name="vertex_count_label">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="8" column="5">
     <widget class="QLabel" name="index_count_label">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
   <zorder>z_rot_btn</zorder>
   <zorder>label</zorder>
   <zorder>y_move_btn</zorder>
   <zorder>x_move_btn</zorder>
   <zorder>y_rot_btn</zorder>
   <zorder>file_name</zorder>
   <zorder>file_browse_btn</zorder>
   <zorder>zoom_in_btn</zorder>
   <zorder>x_rot_btn</zorder>
   <zorder>z_move_btn</zorder>
   <zorder>label_4</zorder>
   <zorder>openGLWidget</zorder>
   <zorder>green_bg</zorder>
   <zorder>blue_bg</zorder>
   <zorder>red_clr_sbox</zorder>
   <zorder>green_clr_sbox</zorder>
   <zorder>blue_clr_sbox</zorder>
   <zorder>parallel_proj_btn</zorder>
   <zorder>central_proj_btn</zorder>
   <zorder>red_edge_sbox</zorder>
   <zorder>green_edge_sbox</zorder>
   <zorder>blue_edge_sbox</zorder>
   <zorder>red_edge_btn</zorder>
   <zorder>green_edge_btn</zorder>
   <zorder>blue_edge_btn</zorder>
   <zorder>width_edge_sbox</zorder>
   <zorder>width_edge_btn</zorder>
   <zorder>dashed_btn</zorder>
   <zorder>straight_btn</zorder>
   <zorder>red_bg</zorder>
   <zorder>x_move_sbox</zorder>
   <zorder>y_move_sbox</zorder>
   <zorder>z_move_sbox</zorder>
   <zorder>x_rot_sbox</zorder>
   <zorder>y_rot_sbox</zorder>
   <zorder>z_rot_sbox</zorder>
   <zorder>zoom_out_btn</zorder>
   <zorder>vertex_count_label</zorder>
   <zorder>index_count_label</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>996</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Render</class>
   <extends>QOpenGLWidget</extends>
   <header>Render.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
